### PR TITLE
auth: add default-catalog-zone setting

### PR DESCRIPTION
**Pull request by Habbie**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Review by rgacogne — APPROVED** [review]

File: pdns/ws-auth.cc [hunk 1]
```diff
@@ -1872,6 +1872,13 @@ static void apiServerZonesPost(HttpRequest* req, HttpResponse* resp) {
 
   updateDomainSettingsFromDocument(B, di, zonename, document, !new_records.empty());
 
+  if (!catalog && (kind != DomainInfo::Producer && kind != DomainInfo::Consumer )) {
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/ws-auth.cc:1875]: alright, that was my other option, will do

[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/ws-auth.cc:1875]: but not `isPrimaryType()`, right :)

[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/ws-auth.cc:1875]: amended

File: docs/settings.rst [hunk 2]
```diff
@@ -366,6 +366,18 @@ The value of :ref:`metadata-api-rectify` if it is not set on the zone.
...
+
+``default-catalog-zone``
+------------------------
+
+- String:
+- Default: empty
+
+.. versionadded:: 4.8.3
+
+When a zone is created via the API, and the request does not specify a catalog zone, the name given here will be used.
```
Comment (mind04) [inline docs/settings.rst:379]:
```suggestion
When a primary zone is created via the API, and the request does not specify a catalog zone, the name given here will be used.
```

File: pdns/auth-main.cc [hunk 3]
```diff
@@ -328,6 +328,8 @@ static void declareArguments()
   ::arg().setSwitch("consistent-backends", "Assume individual zones are not divided over backends. Send only ANY lookup operations to the backend to reduce the number of lookups") = "yes";
 
   ::arg().set("rng", "Specify the random number generator to use. Valid values are auto,sodium,openssl,getrandom,arc4random,urandom.") = "auto";
+
+  ::arg().set("default-catalog-zone", "Catalog zone to assign newly created zones (via the API) to") = "";
```
Comment (mind04) [inline pdns/auth-main.cc:332]:
```suggestion
  ::arg().set("default-catalog-zone", "Catalog zone to assign newly created primary zones (via the API) to") = "";
```

**Review by mind04 — APPROVED** [review]
Two doc suggestion, otherwise fine

**Comment by Habbie** [comment]
thanks @mind04, applied and squashed

**Pull request merged**